go-controller/pkg/factory/factory.go

```diff
@@ -215,6 +203,22 @@ func NewMasterWatchFactory(ovnClientset *util.OVNClientset) (*WatchFactory, erro
 		}
 	}
 
+	// Enable endpointSlice watch only for combined mode master+node
```

Comment:
@zshi-redhat so before we always watched EndpointSlice, right? But now we only do it in combined mode? Was the previous EndpointSlice watch just unused? (I think so, since there's only one hit in node.go, but just checking...)

Reply:
@dcbw  In separate mode (ovnkube-master and ovnkube-node run in separate daemonsets), EndpointSlice is always watched in ovnkube-node watchFactory - [this](https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/factory/factory.go#L304) is not changed and still required for separate mode.
This PR adds another EndpointSlice watch in master watchFactory, which will only be lightened when master and node run in combined mode. In this combined mode, the node watchFactory is unused.